test/trento_web/controllers/v1/settings_controller_test.exs

```diff
@@ -14,4 +15,99 @@ defmodule TrentoWeb.V1.SettingsControllerTest do
     |> json_response(200)
     |> assert_schema("PlatformSettings", api_spec)
   end
+
+  describe "ApiKeySettings" do
+    setup do
+      %{api_spec: ApiSpec.spec()}
+    end
+
+    test "should return not found when api key settings are not configured", %{
+      conn: conn,
+      api_spec: api_spec
+    } do
+      conn
+      |> get("/api/v1/settings/api_key")
+      |> json_response(404)
+      |> assert_schema("NotFound", api_spec)
+    end
+
+    test "should return the api key settings if they are configured", %{
+      conn: conn,
+      api_spec: api_spec
+    } do
+      insert(:api_key_settings)
+
+      conn
+      |> get("/api/v1/settings/api_key")
+      |> json_response(200)
+      |> assert_schema("ApiKeySettings", api_spec)
+    end
+
+    test "should not update the api key settings if they are not configured and return not found",
+         %{conn: conn, api_spec: api_spec} do
+      conn
+      |> put_req_header("content-type", "application/json")
+      |> patch("/api/v1/settings/api_key", %{
+        api_key_expire_at: DateTime.to_iso8601(DateTime.utc_now())
+      })
+      |> json_response(404)
+      |> assert_schema("NotFound", api_spec)
+    end
+
+    test "should update the api key settings if they are configured and return the updated settings",
```

Comment:
`should udpate the api key settings if it is configured returning the updated settings`